mondey_backend/src/mondey_backend/routers/milestones.py

```diff
@@ -67,6 +70,42 @@ def get_milestone_groups(
                 )
             )
         ).all()
-        return milestone_groups
+
+        # compute progress from milestone answers. While this is not
+        # the most efficient way, it is the least intrusive and concentrates
+        # progress reporting in the least amount of code - this function and the
+        # MilestonAnswerPublic model.
+        # The actual database does not need to know about progress - it's
+        # a derived value only for being displayed to the user.
```

Comment:
I'd suggest putting this logic in the frontend:
- `progress` is only used by the frontend
- the frontend already has the data needed to calculate it (milestones list and current answer session)
- and it should actually be less code to implement there than here, e.g. something like (untested)
```js
milestones.filter((milestone) => {return `${milestone.id}` in answerSession.answers}).length / milestones.length
```
- this reduces load on the server
- and means we don't need to re-calculate it in the server when the user submits an answer

Reply:
true, we already need the milestone answer session object.

Reply:
was originally hesitant to put more computation into the frontend, b/c it already has quite a bit to do.